addon/components/tag-filter.js

```diff
@@ -20,6 +21,26 @@ export default class TagFilterComponent extends Component {
     return this.tags.fetchSearchTags.perform(this.args.category);
   });
 
+  activeMarks = trackedFunction(this, async () => {
+    const activeMarks = await this.store
+      .peekAll("document")
+      .reduce(async (acc, doc) => {
+        const marks = await doc.marks;
+        acc = await acc;
+        if (
+          (this.args.category === undefined ||
+            (await doc.category) === this.args.category) &&
+          marks.length > 0
+        ) {
+          marks.forEach((mark) => acc.add(mark.id));
+        }
+
+        return acc;
+      }, new Set());
+
+    return this.marks.marks.records?.filter((mark) => activeMarks.has(mark.id));
+  });
```

Comment:
let's pass the currently displayed documents to the tag filter, and check those for used marks (and tags as well!)